packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -34,6 +34,7 @@ export type TransferRoot = {
   bondTransferRootId?: string
   challenged?: boolean
   challengeExpired?: boolean
+  allSettled?: boolean
```

Comment:
I believe we need to set in the event handlers. In the syncWatcher, `handleMultipleWithdrawalsSettledEvent()` loops through all transfer IDs and marks them all `withdrawalBondSettled` if they are settled. I think `allSettled` should be set in this function to `true` if all of those IDs are settled and `false` if not. This will guarantee the state for the initial sync. It also marks them as settled so the settle watcher does not keep trying to settle then updating the DB downstream.

Reply:
Good idea, will add it to the sync handler as well